references/classification/train.py

```diff
@@ -229,12 +229,16 @@ def main(args):
 
     criterion = nn.CrossEntropyLoss(label_smoothing=args.label_smoothing)
 
-    if args.norm_weight_decay is None:
-        parameters = model.parameters()
-    else:
-        param_groups = torchvision.ops._utils.split_normalization_params(model)
-        wd_groups = [args.norm_weight_decay, args.weight_decay]
-        parameters = [{"params": p, "weight_decay": w} for p, w in zip(param_groups, wd_groups) if p]
+    custom_keys_weight_decay = None
+    if hasattr(model, "no_weight_decay_keys"):
+        custom_keys_weight_decay = {k: 0.0 for k in model.no_weight_decay_keys()}
```

Comment:
I don't think we should introduce this type of API in the models. Whenever we attach custom properties in models we risk compatibility with solutions like JIT/FX etc. Is it an option to pass these as arguments?

Reply:
@datumbox in fact, the method `no_weight_decay_keys` is not included in the `forward` function, so I think it has no risk compatibility with solutions like JIT/FX etc. If we pass the `keys` as args, I think we should be careful to pass the argument for each model, maybe some users can forget it?

Reply:
Historically we have been avoiding properties/attributes on the nn.Module and try to keep it barebone. We could discuss pros/cons for this on a separate RFC but I think we shouldn't link this PR to the resolution of it because it will cause delays. I think passing `no_weight_decay_keys` is fine as long as we document them as part of the training command but this depends on how many arguments you have. Could you provide an example? 

An alternative approach could be to add this specifically for Swin/ViT in a utility method in references as a temporary solution, until the RFC is discussed.

Reply:
Reminder that this still need to be removed.